pylint/checkers/base.py

```diff
@@ -2345,7 +2345,9 @@ class ComparisonChecker(_BasicChecker):
     def _check_singleton_comparison(self, singleton, root_node, negative_check=False):
         if singleton.value is True:
             if not negative_check:
-                suggestion = "just 'expr'"
+                suggestion = "just '{}'".format(
+                    "expr" if utils.is_test_condition(root_node) else "bool(expr)"
```

Comment:
I know it's almost a different feature than what you're doing here, but would it be possible to do bool({expr}) by passing expr to `_check_singleton_comparison` ?  Also most of the time it should be just "expr" and the cast to bool is implicit ? I think it would make for a lot better message. what do you think ?

Reply:
From looking at your comment below, by `bool({expr})` do you mean show the actual expression in question instead of `"expr"`?

Reply:
> Also most of the time it should be just "expr" and the cast to bool is implicit ?

What do you mean by this?

Reply:
Yes sorry I meant the actual expression. For example if the real expression is "x", I think we can also output `Consider replacing 'x == True' to just 'x' instead` and not `Consider replacing 'x == True' to just 'bool(x)'` when python would call cast to a boolean ?

Reply:
That's what this PR was originally addressing -- situations where Python wouldn't cast to boolean, i.e. assigning to a variable as mentioned in the description. Am I understanding you correctly? But yes, I agree that the above would make for a much better error message, I'll add that in 👍